### PR TITLE
haskellPackages.lzma: loosen version bounds for test dependencies

### PR DESCRIPTION
**Pull request by adnelson**
###### Motivation for this change

The Haskell `lzma` library refuses to build, complaining that HUnit and QuickCheck are outside of the given version bounds.

###### Things done

One option would be to disable the unit tests, but the tests pass just fine with the versions in nixpkgs, so I just patched the cabal file to loosen the bounds.

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -566,6 +566,9 @@ self: super: {
   # https://github.com/alphaHeavy/lzma-enumerator/issues/3
   lzma-enumerator = dontCheck super.lzma-enumerator;
 
+  # Unpin some versions
+  lzma = appendPatch super.lzma ./patches/lzma-tests.patch;
```
Comment (peti) [inline pkgs/development/haskell-modules/configuration-common.nix]: Could you please report this issue to @hvr at https://github.com/hvr/lzma/issues and add the link to that issue into the comment here? That makes it easier for other people to decide in the future whether the issue still exists.

**Review by peti — CHANGES_REQUESTED** [review]

File: pkgs/development/haskell-modules/configuration-common.nix (hunk 1 above)
Comment (adnelson) [reply pkgs/development/haskell-modules/configuration-common.nix]: https://github.com/haskell-hvr/lzma/issues/8

Comment (peti) [reply pkgs/development/haskell-modules/configuration-common.nix]: Please add that URL into the source code; it should replace the "Unpin some versions" comment.

**Comment by adnelson** [comment]
@peti fixed, sorry for the delay

**Pull request merged**